pages/notifications.tsx

```diff
@@ -0,0 +1,44 @@
+import Head from 'next/head'
+
+import { useClerk } from '@clerk/clerk-react'
+import { withServerSideAuth } from '@clerk/nextjs/ssr'
+import getUserDetails from '../helpers/user-details'
+import Page from '../components/Page'
+import { ServerPageProps } from '../types/types'
+import { Box, Text } from 'grommet'
+const Notifications = ({ user }: ServerPageProps) => {
+  return (
+    <>
```

Comment:
Have you thought about creating a general pages component that takes in a page title + content props. That way we don't need to keep repeating the basic structure for each page (head, page, box components) and more importantly don't need to keep repeating the getServerSideProps component?

Reply:
This is the right way of thinking about it, however, this is where `Next.js` Differs from React client apps. we need to have the `getServerSideProps` call on each page so that we can generate it on the server in the proper JAMStack way! This might seem less modular. We can only call `getServerSideProps` on `/pages/*` because all the components downstream will be statically rendered regardless. So your intuition is correct generally, but next works a little differently. See the documentation for [`getServerSideProps`](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props)